### ISSUE TITLE
[azappconfig] NewListSettingsPager: Paging to second page fails ("no Host in request URL")

### ISSUE THREAD

**Issue by FloppyNotFound** [Azure/azure-sdk-for-go#19415]
#### System Info
- import path of package in question: "github.com/Azure/azure-sdk-for-go/sdk/data/azappconfig" @ v0.4.2
- azcore@v1.1.1
- output of `go version`: go1.18.6 windows/amd64

#### Report
- What happened?
I have an Azure App Configuration instance with 115 values in it. I am trying to get a complete list of these entries by using the NewListSettingsPager and paging through the results.
The first request gets me the first 100 entries without any issues. 
When trying to load the second result page of the NewListSettingsPager, an error is thrown: "no Host in request URL"

- What did you expect or want to happen?
The next (and last) 15 results are provided in the next page.

- How can we reproduce it?

1. Create an instance of Azure App Configuration and insert 115 configuration items.
2. Create a Go application, install the azappconfig SDK and try to load the results by using the following method:
```
func getAllConfigItems(ctx *gin.Context) ([]azappconfig.Setting, error) {
	appConfigTicketStatusConnectionString := "Endpoint=https://<endpoint>.azconfig.io;Id=<id>;Secret=<secret>"
	client, err := azappconfig.NewClientFromConnectionString(appConfigTicketStatusConnectionString, nil)

	if err != nil {
		return nil, errors.New("could not load ticket status app config")
	}

	selector := azappconfig.SettingSelector{}
	options := azappconfig.ListSettingsOptions{}
	pager := client.NewListSettingsPager(selector, &options)

	settings := []azappconfig.Setting{}

	hasMore := true
	for ok := true; ok; ok = hasMore {

		page, err := pager.NextPage(ctx)
		if err != nil {
			return nil, err
		}

		settings = append(settings, page.Settings...)

		hasMore = pager.More()
	}

	return settings, nil
}
```

I have tracked down the issue to the following method, in which the current page is set after successfully receiving the first 100 results. The response seems to have a NextLink, which indeed does not have a Host name set:
![grafik](https://user-images.githubusercontent.com/18663555/197763980-8d0e8b6e-6cb4-4427-aafa-a6d95942bbec.png)

Later, when trying to fetch the second result page, and therefore a new requst is initiated, the request fails here, because of the missing Host:
![grafik](https://user-images.githubusercontent.com/18663555/197764703-34b5870d-b93c-4299-9db9-4f9ad253bb2a.png)

Am I missing something, or is there even an easier way to get a complete list of (non-paginated) items?
Thanks in advance.

**Comment by jhendrixMSFT** [comment]
Thanks for the detailed bug report, this is helpful.

**Comment by jhendrixMSFT** [comment]
According to the [docs](https://learn.microsoft.com/en-us/azure/azure-app-configuration/rest-api-key-value#pagination) the `nextLink` value is a relative URI, so we need to join it with the client's host.

Fix will be in `v0.4.3` which will be released first week of November.
